### ISSUE TITLE
RecordFieldNamesMustBePascalCase

### ISSUE THREAD

**Issue by sylvanc** [fsprojects/FSharpLint#180]
### Description

Using FSharpLint via Ionide/fsi in vscode, I was creating a record that needed to be compatible with a C# object. I received the linter message "Expected a Pascal case identifier", via the rule `RecordFieldNamesMustBePascalCase`.

This is perhaps a bit strong, as record field names need not be Pascal case, and for compatibility with an existing C# object, occasionally may need to _not_ be.

### Repro steps

```fsharp
type Response = {
    body: string
}
```

### Expected behavior

Either no message, or a less strident one.

### Actual behavior

> Expected a Pascal case identifier, found `body`.
> Response.body: string

### Known workarounds

Ignore the resulting message.

### Related information 

* Operating system: cross-platform
* Branch
* .NET Runtime, CoreCLR or Mono Version: cross-runtime
* Performance information, links to performance testing scripts

**Comment by dsyme** [comment]
@duckmatt See also my comment here: https://github.com/ionide/ionide-vscode-fsharp/issues/148#issuecomment-246694189

**Comment by duckmatt** [comment]
Thanks for the report, this feedback is very useful. I've continued the discussion in #174 

As an aside, wouldn't compatibility with C# usually use pascal case as that's the standard convention for C# properties?

**Comment by sylvanc** [comment]
Yes, I agree, usually it would be :) This particular use case is calling `HttpRequestMessageExtensions.CreateResponse<obj>` where C# is expecting to find a `body` field in the attached `obj` dynamically.

**Comment by duckmatt** [comment]
Ah I see, that's an interesting scenario. The message is now "Consider changing `body` to PascalCase", and the rule is also off by default :)

**Comment by sylvanc** [comment]
Great, thanks!
